### ISSUE TITLE
Firmware update

### ISSUE THREAD

**Issue by ste34144** [jamesshao8/portapack-havoc-modified#22]
Is there any possibility to obtain the last firmware for my h3 se mini? I want to upgrade with last feature you add. Thank you.

**Comment by jamesshao8** [comment]
[portapack-h3_with_add_on-shao.zip](https://github.com/jamesshao8/portapack-havoc-modified/files/11858358/portapack-h3_with_add_on-shao.zip)
v1.1.0

**Comment by ste34144** [comment]
Really thank you. Can you also send a guide for this model to upgrade firmware or where can I read how to make the upgrade? Thank you so much for the response!!

**Comment by jamesshao8** [comment]
> Really thank you. Can you also send a guide for this model to upgrade firmware or where can I read how to make the upgrade? Thank you so much for the response!!

You can switch to HackRF Mode which looks like an APP on the bottom of the Menu. Then flash the firmware using hackrf_spiflash command, which comes from HackRF driver. The steps for flashing its firmware are the same as an single HackRF board.

If you are not familiar with Ubuntu or other linux distro. Please send me an email: 13671890179@139.com 
I will send you other tools.

**Comment by jamesshao8** [comment]
You can switch to HackRF Mode which looks like an APP on the bottom of the Menu. Then flash the firmware using hackrf_spiflash command, which comes from HackRF driver. The steps for flashing its firmware are the same as an single HackRF board.

If you are not familiar with Ubuntu or other linux distro. Please send me an email: ***@***.*** 
I will send you other tools.









--------------------------------------------------------------------------------





------------------ 原始邮件 ------------------




 
 发件人: ste34144  ***@***.***>
 

 
 发送时间: 2023-06-25 09:08:29
 

 
 收件人:jamesshao8/portapack-havoc-modified ***@***.***> 

 
 抄送:shao ***@***.***>,Comment ***@***.***> 

 
 主题: Re: [jamesshao8/portapack-havoc-modified] Firmware update (Issue #22)
 





 Really thank you. Can you also send a guide for this model to upgrade firmware or where can I read how to make the upgrade? Thank you so much for the response!!

**Comment by ste34144** [comment]
Thank you. Too gentle. Ok I’ll follow the instructions. It seems a simple thing. Really grateful. Now I’ll try this method.

**Comment by ste34144** [comment]
Thanks it works! Wonderful! Thank you so much!

**Comment by jamesshao8** [comment]
> Thanks it works! Wonderful! Thank you so much!

You are welcome.

From now on, I will upload new firmware here:
https://github.com/jamesshao8/portapack-havoc-modified/releases/tag/v1.1.5

I fixed some bugs in walkie-talkie and landscape view.

**Comment by ste34144** [comment]
Yeah,awesome! You're great! Thank you!

**Comment by aut0li** [comment]
Any new firmware lately? Thanks.

**Comment by jamesshao8** [comment]
Hi there,The latest version is v1.1.5. I know it’s been a while since last update. But I am busy designing a new case.Regards,Shao在 2023年8月30日，20:21，aut0li ***@***.***> 写道：﻿
Any new firmware lately? Thanks.

**Comment by ste34144** [comment]
Any news for the update of the firmware sir? Thank you so much!

**Comment by jamesshao8** [comment]
Hi there,It is planned to announce a new update at the beginning of 2024.Regards,Shao在 2023年12月1日，04:32，ste34144 ***@***.***> 写道：﻿
Any news for the update of the firmware sir? Thank you so much!

**Comment by ste34144** [comment]
Thank you so much. The ultimate firmware, update it's amazing! Really awesome! THANKS FOR ALL!

**Comment by GabrielC0** [comment]
Bonjour, je possède un Portapak h3 mini et j'aimerai savoir si la dernier version du firmware est bien le V1.28 et y a il un mode emploi détailler sur les différentes possibilités qu'offre cette machine ? merci a vous

**Comment by GabrielC0** [comment]
Est il possible aussi de mettre à jour le mode "replay" car lorsque nous avons plusieurs fichier a rejouer, nous sommes obligées de retourner dans le fichier racine et de parcourir tout ls fichiers. Je souhaiterai un moyen de retourner dans le dernier dossier du fichier lu s'il vous plait. merci infiniment du travail

**Comment by jamesshao8** [comment]
> Bonjour, je possède un Portapak h3 mini et j'aimerai savoir si la dernier version du firmware est bien le V1.28 et y a il un mode emploi détailler sur les différentes possibilités qu'offre cette machine ? merci a vous

Hello,

Sorry, I don't understand French. The latest firmware is v1.31 now. You can download firmware file from release page. Do you have an email address so I can send you the user manual?

**Comment by jamesshao8** [comment]
> Est il possible aussi de mettre à jour le mode "replay" car lorsque nous avons plusieurs fichier a rejouer, nous sommes obligées de retourner dans le fichier racine et de parcourir tout ls fichiers. Je souhaiterai un moyen de retourner dans le dernier dossier du fichier lu s'il vous plait. merci infiniment du travail

Do you mean that after one file is replayed, you want the file browser to navigate to the same folder you have used last time, instead of going to the root folder?

**Comment by GabrielC0** [comment]
> > Bonjour, je possède un Portapak h3 mini et j'aimerai savoir si la dernier version du firmware est bien le V1.28 et y a il un mode emploi détailler sur les différentes possibilités qu'offre cette machine ? merci a vous
> 
> 
> 
> Hello,
> 
> 
> 
> Sorry, I don't understand French. The latest firmware is v1.31 now. You can download firmware file from release page. Do you have an email address so I can send you the user manual?


Hello, I can speak English if you want. Please send me the user manual in this email : Gabriel.changrenier@Icloud.com 
Thanks for your help and have a good day.
Gabriel

**Comment by GabrielC0** [comment]
> > Est il possible aussi de mettre à jour le mode "replay" car lorsque nous avons plusieurs fichier a rejouer, nous sommes obligées de retourner dans le fichier racine et de parcourir tout ls fichiers. Je souhaiterai un moyen de retourner dans le dernier dossier du fichier lu s'il vous plait. merci infiniment du travail
> 
> 
> 
> Do you mean that after one file is replayed, you want the file browser to navigate to the same folder you have used last time, instead of going to the root folder?

Yes exactly because when we have many files to replayed, we need to go back in the folder after any replays.
Thanks

**Comment by jamesshao8** [comment]
> > > Est il possible aussi de mettre à jour le mode "replay" car lorsque nous avons plusieurs fichier a rejouer, nous sommes obligées de retourner dans le fichier racine et de parcourir tout ls fichiers. Je souhaiterai un moyen de retourner dans le dernier dossier du fichier lu s'il vous plait. merci infiniment du travail
> > 
> > 
> > Do you mean that after one file is replayed, you want the file browser to navigate to the same folder you have used last time, instead of going to the root folder?
> 
> Yes exactly because when we have many files to replayed, we need to go back in the folder after any replays. Thanks

Understood. I will try to tweak that, and will let you know once it's done.

**Comment by GabrielC0** [comment]
Hey, i have a sugestion for update. its possible to have a counter when i replay frequences he increse of one. its juste a counter to show the number of the replay in all.
Thanks all

**Comment by jamesshao8** [comment]
> > > > Est il possible aussi de mettre à jour le mode "replay" car lorsque nous avons plusieurs fichier a rejouer, nous sommes obligées de retourner dans le fichier racine et de parcourir tout ls fichiers. Je souhaiterai un moyen de retourner dans le dernier dossier du fichier lu s'il vous plait. merci infiniment du travail
> > > 
> > > 
> > > Do you mean that after one file is replayed, you want the file browser to navigate to the same folder you have used last time, instead of going to the root folder?
> > 
> > 
> > Yes exactly because when we have many files to replayed, we need to go back in the folder after any replays. Thanks
> 
> Understood. I will try to tweak that, and will let you know once it's done.

I have just read the implementation of FileLoadView. I think there isn't easy way to directly go to a sub directory other than the root folder.

**Comment by jamesshao8** [comment]
> Hey, i have a sugestion for update. its possible to have a counter when i replay frequences he increse of one. its juste a counter to show the number of the replay in all. Thanks all

I used to develop a special firmware that will iterate files in GPS Sim App and automatically send them sequentially. I am not sure if you want the same for Replay App. I think this is a very special requirement and won't be necessary for other users.

**Comment by GabrielC0** [comment]
Hello, where I can Tcheck if my portapack is in the last version please ?

**Comment by jamesshao8** [comment]
> Hello, where I can Tcheck if my portapack is in the last version please ?

Hello Gabriel,

The version number is on the bottom-left corner of the main menu.
Currently, the latest version is v1.35.

**Comment by GabrielC0** [comment]
Thank you for the information, where can I follow the latest updates to avoid asking the question every time?

**Comment by jamesshao8** [comment]
Hi,




The latest firmware is published in release page. For example, it's version v1.35 now, and you can download the 135.zip file from there. You can get the *.bin file inside it.




The flashing method isn't changed, you can copy and paste the latest bin file to your original flashing program folder and do the same as last time.

Regards,

Shao






--------------------------------------------------------------------------------





------------------ 原始邮件 ------------------




 
 发件人: GabrielC0  ***@***.***>
 

 
 发送时间: 2024-08-27 04:43:55
 

 
 收件人:jamesshao8/portapack-havoc-modified ***@***.***> 

 
 抄送:shao ***@***.***>,Comment ***@***.***> 

 
 主题: Re: [jamesshao8/portapack-havoc-modified] Firmware update (Issue #22)
 





 Thank you for the information, where can I follow the latest updates to avoid asking the question every time?

**Comment by GabrielC0** [comment]
Thanks ! Is it really impossible that when I replay a .c16 file to return to the previous file and not to the root file?

**Comment by jamesshao8** [comment]
> Thanks ! Is it really impossible that when I replay a .c16 file to return to the previous file and not to the root file?

If you don't like the root folder. I can create a special folder, so that capture/replay works inside that folder. But the folder name will be fixed instead of automatically picked up from previous file. Does it work for you?

**Comment by GabrielC0** [comment]
YESS !! Thanks

**Comment by jamesshao8** [comment]
> YESS !! Thanks

I sent you a special firmware through email.

**Comment by GabrielC0** [comment]
Yes thanks, Is it possible to have a replay file counter? If it's not possible it's ok

**Comment by jamesshao8** [comment]
> Yes thanks, Is it possible to have a replay file counter? If it's not possible it's ok

Hi, Nice to hear it's useful for you. But I don't plan to add the counter, because it's not a common requirement for other users. Right now, I have many priorities to do.

**Comment by Worlood-josha** [comment]
Has the firmware stopped receiving updates?

**Comment by jamesshao8** [comment]
Hi,This firmware is quite stable now. It’s still open for bug fixes or other useful updates, but won’t be updated as frequent as before.I am busy working on a new product and it will be published soon.在 2025年6月1日，03:24，Worlood-josha ***@***.***> 写道：﻿Worlood-josha left a comment (jamesshao8/portapack-havoc-modified#22)
Has the firmware stopped receiving updates?
